request/graphql/schema/collection.go

```diff
@@ -159,15 +132,211 @@ func fromAstDefinition(
 		return fieldDescriptions[i].Name < fieldDescriptions[j].Name
 	})
 
+	for _, directive := range def.Directives {
+		if directive.Name.Value == indexDirectiveLabel {
+			index, err := indexFromAST(directive)
+			if err != nil {
+				return client.CollectionDescription{}, err
+			}
+			indexDescriptions = append(indexDescriptions, index)
+		}
+	}
+
 	return client.CollectionDescription{
 		Name: def.Name.Value,
 		Schema: client.SchemaDescription{
 			Name:   def.Name.Value,
 			Fields: fieldDescriptions,
 		},
+		Indexes: indexDescriptions,
 	}, nil
 }
 
+func isValidIndexName(name string) bool {
```

Comment:
todo: I think this is worth documenting, both for us internal developers, and for external users of the client package.